Bug: Download of collection failed - 1.6 AetherValley

Attempted to download collection: https://next.nexusmods.com/stardewvalley/collections/e1b4ep

**Repro steps:**
1.  Add collection
2. View nexus mods app
3. Click ok on uncaught exception error

**Expected result:**
Collection downloads

**Actual result:**
Collection download stalls on 76 of 119

![Image](https://github.com/user-attachments/assets/cd162da8-8486-4f0c-9e04-0dab9ae845c1)

EDIT (AL):
## Stack trace:
<details>

```
00:07:11.788 [ERROR] Job 75ff1e96-0244-43a4-a423-a4b4e4824b48 of type NexusMods.Collections.DownloadCollectionJob failed|System.Threading.Tasks.TaskCanceledException: The request was canceled due to the configured HttpClient.Timeout of 5 seconds elapsing.
 ---> System.TimeoutException: The operation was canceled.
 ---> System.Threading.Tasks.TaskCanceledException: The operation was canceled.
 ---> System.IO.IOException: Unable to read data from the transport connection: The I/O operation has been aborted because of either a thread exit or an application request..
 ---> System.Net.Sockets.SocketException (995): The I/O operation has been aborted because of either a thread exit or an application request.
   --- End of inner exception stack trace ---
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
   at System.Net.Security.SslStream.EnsureFullTlsFrameAsync[TIOAdapter](CancellationToken cancellationToken, Int32 estimatedSize)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](Memory`1 buffer, CancellationToken cancellationToken)
   at System.Runtime.CompilerServices.PoolingAsyncValueTaskMethodBuilder`1.StateMachineBox`1.System.Threading.Tasks.Sources.IValueTaskSource<TResult>.GetResult(Int16 token)
   at System.Net.Http.HttpConnection.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Resilience.ResilienceHandler.<>c.<<SendAsync>b__3_0>d.MoveNext()
--- End of stack trace from previous location ---
   at Polly.Outcome`1.ThrowIfException()
   at Microsoft.Extensions.Http.Resilience.ResilienceHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   --- End of inner exception stack trace ---
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpClient.HandleFailure(Exception e, Boolean telemetryStarted, HttpResponseMessage response, CancellationTokenSource cts, CancellationToken cancellationToken, CancellationTokenSource pendingRequestsCts)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
   at NexusMods.Networking.HttpDownloader.HttpDownloadJob.StartAsync(IJobContext`1 context) in C:\Dev\Repos\NexusMods.App\src\Networking\NexusMods.Networking.HttpDownloader\HttpDownloadJob.cs:line 116
   at NexusMods.Networking.HttpDownloader.HttpDownloadJob.StartAsync(IJobContext`1 context) in C:\Dev\Repos\NexusMods.App\src\Networking\NexusMods.Networking.HttpDownloader\HttpDownloadJob.cs:line 175
   at NexusMods.Networking.HttpDownloader.HttpDownloadJob.StartAsync(IJobContext`1 context) in C:\Dev\Repos\NexusMods.App\src\Networking\NexusMods.Networking.HttpDownloader\HttpDownloadJob.cs:line 175
   at NexusMods.Jobs.JobContext`2.Start() in C:\Dev\Repos\NexusMods.App\src\NexusMods.Jobs\JobContext.cs:line 42
   at NexusMods.Networking.NexusWebApi.NexusModsDownloadJob.StartAsync(IJobContext`1 context) in C:\Dev\Repos\NexusMods.App\src\Networking\NexusMods.Networking.NexusWebApi\NexusModsDownloadJob.cs:line 34
   at NexusMods.Jobs.JobContext`2.Start() in C:\Dev\Repos\NexusMods.App\src\NexusMods.Jobs\JobContext.cs:line 42
   at NexusMods.Library.AddDownloadJob.StartAsync(IJobContext`1 context) in C:\Dev\Repos\NexusMods.App\src\NexusMods.Library\AddDownloadJob.cs:line 32
   at NexusMods.Jobs.JobContext`2.Start() in C:\Dev\Repos\NexusMods.App\src\NexusMods.Jobs\JobContext.cs:line 42
   at NexusMods.Collections.CollectionDownloader.Download(ReadOnly download, CancellationToken cancellationToken) in C:\Dev\Repos\NexusMods.App\src\NexusMods.Collections\CollectionDownloader.cs:line 164
   at NexusMods.Collections.CollectionDownloader.Download(ReadOnly download, CancellationToken cancellationToken) in C:\Dev\Repos\NexusMods.App\src\NexusMods.Collections\CollectionDownloader.cs:line 164
   at NexusMods.Collections.DownloadCollectionJob.<>c__DisplayClass20_0.<<StartAsync>b__0>d.MoveNext() in C:\Dev\Repos\NexusMods.App\src\NexusMods.Collections\DownloadCollectionJob.cs:line 33
--- End of stack trace from previous location ---
   at System.Threading.Tasks.Parallel.<>c__49`1.<<ForAsync>b__49_0>d.MoveNext()
--- End of stack trace from previous location ---
   at NexusMods.Collections.DownloadCollectionJob.StartAsync(IJobContext`1 context) in C:\Dev\Repos\NexusMods.App\src\NexusMods.Collections\DownloadCollectionJob.cs:line 21
   at NexusMods.Jobs.JobContext`2.Start() in C:\Dev\Repos\NexusMods.App\src\NexusMods.Jobs\JobContext.cs:line 42
```

</details>

--- a/src/Networking/NexusMods.Networking.HttpDownloader/HttpDownloadJob.cs
+++ b/src/Networking/NexusMods.Networking.HttpDownloader/HttpDownloadJob.cs
@@ -261,7 +261,7 @@ public record HttpDownloadJob : IJobDefinitionWithStart<HttpDownloadJob, Absolut
         {
             InnerHandler = new SocketsHttpHandler
             {
-                ConnectTimeout = TimeSpan.FromSeconds(10),
+                ConnectTimeout = TimeSpan.FromSeconds(30),
                 KeepAlivePingPolicy = HttpKeepAlivePingPolicy.WithActiveRequests,
                 KeepAlivePingDelay = TimeSpan.FromSeconds(5),
                 KeepAlivePingTimeout = TimeSpan.FromSeconds(20),
@@ -270,7 +270,7 @@ public record HttpDownloadJob : IJobDefinitionWithStart<HttpDownloadJob, Absolut
 
         var client = new HttpClient(handler)
         {
-            Timeout = TimeSpan.FromSeconds(5),
+            Timeout = TimeSpan.FromSeconds(20),
             DefaultRequestVersion = HttpVersion.Version11,
             DefaultVersionPolicy = HttpVersionPolicy.RequestVersionOrHigher,
         };

--- a/src/Networking/NexusMods.Networking.NexusWebApi/NexusModsDownloadJob.cs
+++ b/src/Networking/NexusMods.Networking.NexusWebApi/NexusModsDownloadJob.cs
@@ -1,4 +1,5 @@
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 using NexusMods.Abstractions.Downloads;
 using NexusMods.Abstractions.Jobs;
 using NexusMods.Abstractions.Library.Models;
@@ -12,26 +13,40 @@ namespace NexusMods.Networking.NexusWebApi;
 
 public class NexusModsDownloadJob : IDownloadJob, IJobDefinitionWithStart<NexusModsDownloadJob, AbsolutePath>
 {
+    public required ILogger Logger { private get; init; }
     public required IJobTask<HttpDownloadJob, AbsolutePath> HttpDownloadJob { get; init; }
     public required NexusModsFileMetadata.ReadOnly FileMetadata { get; init; }
 
     /// <inheritdoc/>
     public AbsolutePath Destination => HttpDownloadJob.Job.Destination;
-    
-    public static IJobTask<NexusModsDownloadJob, AbsolutePath> Create(IServiceProvider provider, IJobTask<HttpDownloadJob, AbsolutePath> httpDownloadJob, NexusModsFileMetadata.ReadOnly fileMetadata)
+
+    public static IJobTask<NexusModsDownloadJob, AbsolutePath> Create(
+        IServiceProvider provider,
+        IJobTask<HttpDownloadJob, AbsolutePath> httpDownloadJob,
+        NexusModsFileMetadata.ReadOnly fileMetadata)
     {
         var monitor = provider.GetRequiredService<IJobMonitor>();
         var job = new NexusModsDownloadJob
         {
+            Logger = provider.GetRequiredService<ILogger<NexusModsDownloadJob>>(),
             HttpDownloadJob = httpDownloadJob,
             FileMetadata = fileMetadata,
         };
+
         return monitor.Begin<NexusModsDownloadJob, AbsolutePath>(job);
     }
-    
+
     public async ValueTask<AbsolutePath> StartAsync(IJobContext<NexusModsDownloadJob> context)
     {
-        return await HttpDownloadJob;
+        try
+        {
+            return await HttpDownloadJob;
+        }
+        catch (Exception e)
+        {
+            Logger.LogError(e, "Exception while downloading `{Url}`", FileMetadata.GetUri());
+            throw;
+        }
     }
 
     /// <inheritdoc/>

